kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,222 @@
+package org.eclipse.kura.nm;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.freedesktop.NetworkManager;
+import org.freedesktop.dbus.DBusPath;
+import org.freedesktop.dbus.connections.impl.DBusConnection;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.dbus.interfaces.Properties;
+import org.freedesktop.dbus.types.UInt32;
+import org.freedesktop.networkmanager.Device;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class NMDbusConnectorTest {
+	static final DBusConnection dbusConnection = Mockito.mock(DBusConnection.class, Mockito.RETURNS_SMART_NULLS);
+	static final NetworkManager mockedNetworkManager = Mockito.mock(NetworkManager.class);
+	static NMDbusConnector instanceNMDbusConnector;
+	DBusConnection dbusConnectionInternal;
+	
+	Boolean hasDBusExceptionBeenThrown = false;
+	
+	List<String> internalStringList;
+	
+	@BeforeClass
+	public static void setUpPersistantMocks() throws DBusException {
+		givenBasicMockedDbusConnector();
+	}
```

Comment:
Can you explain why are you doing this?

Setting this fields `static` make them **retain their values between tests**. I understand that you create a new instance with the `@BeforeClass` method but **that's what JUnit is supposed to do out-of-the-box**.

You're opening yourself to side-effects that would make your tests to be a pain to debug and I don't understand why.

Can't you have:

```suggestion
	static final DBusConnection dbusConnection = Mockito.mock(DBusConnection.class, Mockito.RETURNS_SMART_NULLS);
	NetworkManager mockedNetworkManager;
	NMDbusConnector instanceNMDbusConnector;
	DBusConnection dbusConnectionInternal;
	
	Boolean hasDBusExceptionBeenThrown = false;
	
	List<String> internalStringList;
```

and create the mock for each tests?

Reply:
Mmmmmh after thinking about it my assumption might be wrong.

Anyway... i might have an idea on how to fix this.